tc/core/polyhedral/scop.cc

```diff
@@ -456,6 +539,24 @@ detail::ScheduleTree* Scop::tileOuterBand(const TilingView& tileSizes) {
   return res;
 }
 
+std::vector<detail::ScheduleTree*> Scop::tileOuterCoincidentBands(
+    const TilingView& tileSizes) {
+  using namespace tc::polyhedral::detail;
+  auto bands = obtainOuterCoincidentBands(scheduleRoot(), scheduleRoot());
```

Comment:
one useful sanity check here seems to be that no 2 bands you return are nested. It does not seem superfluous to add this check (and maybe also a check that all paths are covered if I understood properly).